velocity/src/main/java/me/leoko/advancedban/velocity/VelocityMethods.java

```diff
@@ -0,0 +1,368 @@
+
+    @Override
+    public String[] getKeys(Object file, String path) {
+        return ((Yaml) file).getSection(path).singleLayerKeySet().toArray(new String[0]);
+    }
+
+    @Override
+    public Yaml getConfig() {
+        return config;
+    }
+
+    @Override
+    public Yaml getMessages() {
+        return messages;
+    }
+
+    @Override
+    public Yaml getLayouts() {
+        return layouts;
+    }
+
+    @Override
+    public void setupMetrics() {
+        // Velocity does not support bStats
+    }
+
+    @Override
+    public boolean isBungee() {
+        return true;
+    }
+
+    @Override
+    public String clearFormatting(String text) {
+        return text.replace("§", "&");
+    }
+
+    @Override
+    public VelocityMain getPlugin() {
+        return VelocityMain.get();
+    }
+
+    @Override
+    public File getDataFolder() {
+        return getPlugin().folder;
+    }
+
+    @Override
+    public void setCommandExecutor(String cmd, TabCompleter tabCompleter) {
+        getPlugin().server.getCommandManager().register(new CommandReceiverVelocity(cmd), cmd);
+    }
+
+    @Override
+    public void sendMessage(Object player, String msg) {
+        if (player instanceof Player) {
+            ((Player) player).sendMessage(TextComponent.of(msg));
+        } else {
+            ((CommandSource) player).sendMessage(TextComponent.of(msg));
+        }
+    }
+
+    @Override
+    public String getName(Object player) {
+        if (player instanceof Player) {
+            return ((Player) player).getUsername();
+        } else {
+            return "CONSOLE";
+        }
+    }
+
+    @Override
+    public String getName(String uuid) {
+        return getPlugin().server.getPlayer(uuid).get().getUsername();
+    }
+
+    @Override
+    public String getIP(Object player) {
+        return ((Player)player).getRemoteAddress().getAddress().getHostAddress();
+    }
+
+    @Override
+    public String getInternUUID(Object player) {
+        return player instanceof  Player ? ((Player) player).getUniqueId().toString().replace("-", "") : "none";
+    }
+
+    @Override
+    public String getInternUUID(String player) {
+        Optional<Player> optionalPlayer = getPlugin().server.getPlayer(player);
+        return optionalPlayer.map(value -> value.getUniqueId().toString().replace("-", "")).orElse(null);
+    }
+
+    @Override
+    public boolean hasPerms(Object player, String perms) {
+        if (player instanceof Player) {
+            return ((Player) player).hasPermission(perms);
+        } else {
+            return ((CommandSource) player).hasPermission(perms);
+        }
+    }
+
+    @Override
+    public boolean hasOfflinePerms(String name, String perms) {
+        return false;
+    }
+
+    @Override
+    public boolean isOnline(String name) {
+        return getPlugin().server.getPlayer(name).isPresent();
+    }
+
+    @Override
+    public Object getPlayer(String name) {
+        return getPlugin().server.getPlayer(name).get();
```

Comment:
Using `Optional#get` changes the behaviour of this method from other MethodInterface implementations, making this one throw NoSuchElementException for nonexistent values. Also, the overridden signature can be changed to `Player`.